config_test.go

```diff
@@ -58,6 +58,28 @@ func TestParseWrongDSN(t *testing.T) {
 	}
 }
 
+func TestParseWrongDNSParams(t *testing.T) {
+	case1 := map[string][]string{
+		"enable_http_compression": {"foo"},
+	}
+	case2 := map[string][]string{
+		"timeout": {"50"},
+	}
+	case3 := map[string][]string{
+		"database": {"whatever"},
+	}
+
+	testCases := []map[string][]string{
+		case1, case2, case3,
+	}
+
+	for _, tc := range testCases {
+		cfg := NewConfig()
+		err := parseDSNParams(cfg, tc)
+		assert.Error(t, err, nil)
```

Comment:
the third argument nil is extra

Reply:
need to check changes, which you made in patch, seems like this test pass on current code, without patch.
```
assert.Error(t, err, "failed to parse param %s=%v, %+v", name, value, "error text")
// or
if assert.Error(t, err, "failed to parse param %s=%v, %+v", name, value, "error text") {
    assert. Contains(t, err.Error(), "failed to parse param " +  name)
}
```

Reply:
done